File: frontend/src/types/report.ts
export interface DayPoint {
  date: string;        // YYYY-MM-DD
  revenue: number;     // centavos
  cost: number;        // centavos
  orders: number;
  units: number;
}

export interface StatusCount {
  status: string;
  count: number;
  total: number;       // centavos
}

export interface SalesReport {
  revenue: number;
  order_count: number;
  units: number;
  avg_ticket: number;
  by_day: DayPoint[];
  by_status: StatusCount[];
}

export interface PurchasesReport {
  total_cost: number;
  purchase_count: number;
  units: number;
  by_day: DayPoint[];
}

export interface MarginProductRow {
  product_id: string;
  title: string;
  units: number;
  revenue: number;
  cogs: number;
  profit: number;
  margin_pct: number | null;
}

export interface MarginsReport {
  revenue: number;
  cogs: number;
  gross_profit: number;
  margin_pct: number | null;
  items_without_cost: number;
  by_product: MarginProductRow[];
}

export interface ReportOverview {
  date_from: string;
  date_to: string;
  sales: SalesReport;
  purchases: PurchasesReport;
  margins: MarginsReport;
}

File: backend/app/schemas/report.py
from datetime import date

from pydantic import BaseModel


class DayPoint(BaseModel):
    date: date
    revenue: int = 0        # centavos
    cost: int = 0           # centavos
    orders: int = 0
    units: int = 0


class StatusCount(BaseModel):
    status: str
    count: int
    total: int              # centavos


class SalesReport(BaseModel):
    revenue: int            # centavos, sólo pedidos en estados de venta efectiva
    order_count: int
    units: int
    avg_ticket: int         # centavos
    by_day: list[DayPoint]
    by_status: list[StatusCount]


class PurchasesReport(BaseModel):
    total_cost: int         # centavos
    purchase_count: int
    units: int
    by_day: list[DayPoint]


class MarginProductRow(BaseModel):
    product_id: str
    title: str
    units: int
    revenue: int            # centavos
    cogs: int               # centavos
    profit: int             # centavos
    margin_pct: float | None


class MarginsReport(BaseModel):
    revenue: int            # centavos
    cogs: int               # centavos
    gross_profit: int       # centavos
    margin_pct: float | None
    items_without_cost: int
    by_product: list[MarginProductRow]


class ReportOverview(BaseModel):
    date_from: date
    date_to: date
    sales: SalesReport
    purchases: PurchasesReport
    margins: MarginsReport
